io/read.go

```diff
@@ -53,3 +55,14 @@ func consistentReadSync(filename string, attempts int, sync func(int)) ([]byte,
 	}
 	return nil, fmt.Errorf("could not get consistent content of %s after %d attempts", filename, attempts)
 }
+
+// ReadAtMost reads up to `limit` bytes from `r`, and reports an error
+// when `limit` bytes are read.
+func ReadAtMost(r io.Reader, limit int64) ([]byte, error) {
+	limitedReader := &io.LimitedReader{R: r, N: limit}
+	data, err := ioutil.ReadAll(limitedReader)
```

Comment:
I think you should check `err` before checking the read limit.

Reply:
Done!